File: level2/N개의_최소공배수/Solution.java
class Solution {
    public int solution(int[] arr) {
        for (int i = 1; i < arr.length; i++) {
            // 최소 공배수 공식 사용
            arr[i] = arr[i - 1] * arr[i] / gcd(arr[i - 1], arr[i]);
        }
        int answer = arr[arr.length - 1];
        return answer;
    }

    // 최대공약수를 구하는 메소드
    private int gcd(int a, int b) {
        return a % b > 0 ? gcd(b, a % b) : b;
    }

    public static void main(String[] args) {
        Solution s = new Solution();
        System.out.println(s.solution(new int[] {2, 6, 8, 14}));
        System.out.println(s.solution(new int[] {1, 2, 3}));
    }
}

File: level2/N개의_최소공배수/solution.js
function solution(arr) {
  // 오름차순 정렬하고
  arr.sort((a, b) => a - b);
  // 제일 큰수
  let max = arr.pop();
  // max가 나누어 떨어지지 않는 수만 남기고
  arr = arr.filter((v) => max % v > 0);
  var answer = max;
  // max의 배수중에 arr에 남은 수 모두로 나누어 떨어질 때까지 반복
  loop: while (true) {
    for (let n of arr) {
      if (answer % n > 0) {
        answer += max;
        continue loop;
      }
    }
    break;
  }
  return answer;
}

console.log(solution([2, 6, 8, 14]));
console.log(solution([1, 2, 3]));
